strapi/design-system
# fix: dts build / pkg exports & husky

### What does it do?

* Fixes the build pipeline for types
* Fixes the package.json export mapping
* Fixes huksy not being installed by default

### Why is it needed?

* in #1273 it turned out the way the ternary was written that we'd always build the types if we omitted the flag which would have introduced a breaking change if done without awareness.
* while working on the helper-plugin I realised that v2 types were not accessible, so I added a specific path for those exports, afterwards they still weren't being picked up I realised in the helper-plugin we need to use node16 module resolution – which showed the issue in the exports map, from what i've learnt doing strapi/strapi#17747 and through other libraries, its _alot_ easier not to have `type: "module"`.
* Husky wasn't working well with yarn 3, I re-followed the README docs of said library

### How to test it?

* Link the DS to helper-plugin
* change Notifications to `tsx`
* edit `tsconfig/client` to have `moduleResolution: "Node16"`
* You should have types for both the root import and the v2 import path
